subprojects/dependency-management/src/main/java/org/gradle/internal/locking/LockFileReaderWriter.java

```diff
@@ -200,6 +200,12 @@ public void writeUniqueLockfile(Map<String, List<String>> lockState) {
         checkValidRoot();
         Path lockfilePath = getUniqueLockfilePath();
 
+        if (lockState.isEmpty()) {
+            // Remove the file when no lock state
+            GFileUtils.deleteQuietly(lockfilePath.toFile());
```

Comment:
Wouldn't we want to have something logged in case something prevents the file from being deleted?

Reply:
I decided not to do it because there is effectively no harm in leaving the file around. Worst case a human has to remove it. While trying to provide clear information may me tricky.